packages/core/src/lambda/activation.ts

```
@@ -76,6 +82,40 @@ export async function activate(context: ExtContext): Promise<void> {
 
         Commands.register('aws.launchDebugConfigForm', async (node: ResourceNode) =>
             registerSamDebugInvokeVueCommand(context.extensionContext, { resource: node })
-        )
+        ),
+
+        Commands.register('aws.appBuilder.tailLogs', async (node: LambdaFunctionNode | TreeNode) => {
+            let functionConfiguration: Lambda.FunctionConfiguration
+            try {
+                const sourceNode = getSourceNode<LambdaFunctionNode>(node)
+                functionConfiguration = sourceNode.configuration
+                const logGroupInfo = {
+                    regionName: sourceNode.regionCode,
+                    groupName: getFunctionLogGroupName(functionConfiguration),
+                }
+
+                const source = isTreeNode(node) ? 'AppBuilder' : 'AwsExplorerLambdaNode'
+                // Show all log streams without having to choose
+                const logStreamFilterData: LogStreamFilterResponse = { type: 'all' }
+                await tailLogGroup(
+                    liveTailRegistry,
+                    source,
+                    liveTailCodeLensProvider,
+                    logGroupInfo,
+                    logStreamFilterData
+                )
+            } catch (err) {
+                if (isError(err as Error, 'ResourceNotFoundException', "LogGroup doesn't exist.")) {
+                    // If we caught this error, then we know `functionConfiguration` actually has a value
+                    throw ToolkitError.chain(
```

will we get this issue when we deploy the function for the first time, and hit taillog before invoking it?

---

For I feel this should be a valid use case

---

Yeah, and that's how the Live Tail API works. It needs an existing log group. We could eventually do something like try once and if it fails we retry (but we might keep retrying forever), but in practice it's better to surface the error, so customers understand it better and retry when needed. Live Tail use case is to "see and filter logs", so it's less likely that people want to use it a lot for a brand new function that doesn't have any invokes yet.